### ISSUE TITLE
Split console from shell components

### ISSUE THREAD

**Issue by piranna** [skerit/janeway#1]
I think the scroll is something generic that can be used in several consoles (it's transparent to the language or the text being shown, think on gnome-terminal scroll functionality or terminator for example), while at the same time the objects expansion is related to the shell (it's mostly vinculed to a particular language, in this case Javascript). This would allow to create a great console and shell for NodeOS (take a look at https://github.com/NodeOS/node-bin-getty and https://github.com/groundwater/node-bin-nsh) :-)

**Comment by skerit** [comment]
Hmm, if I understand correctly you want some kind of abstraction layer between the handling of the log lines and the output on the screen?

Because, yeah, it's very intertwined. It's specifically meant for the terminal, which is why I used the "blessed" curses library.

Or am I misreading something? :)

**Comment by piranna** [comment]
Yes, somewhat :-) It could be done a generic terminal with scroll based on blessed (it has got a lot of improvements in this area) and another part being the shell component itself, with the objects expansion magic and so. I don't know how much could it be possible to make the blessed-on-blessed to work, but the blessed-terminal widget could do the trick according to the demos (it open a new pty... :-) ).
